src/osd/OSD.cc

```diff
@@ -6606,9 +6603,10 @@ void OSD::handle_osd_map(MOSDMap *m)
   write_superblock(t);
   store->queue_transaction(
     service.meta_osr.get(),
-    _t,
-    new C_OnMapApply(&service, _t, pinned_maps, osdmap->get_epoch()),
+    std::move(*_t),
+    new C_OnMapApply(&service, pinned_maps, osdmap->get_epoch()),
     0, 0);
+  delete _t;
```

Comment:
it looks like `_t` could be on the stack, and this funny business above could go away:

```
ObjectStore::Transaction *_t = new ObjectStore::Transaction;
ObjectStore::Transaction &t = *_t;
```